### PR TITLE
feature/utransportmock

### PR DESCRIPTION
**Pull request by debruce**
This is a PR containing code for UTransport mocking and testing of this mock.

### CODE REVIEW

File: test/include/UTransportMock.h [hunk 1]
```diff
@@ -13,5 +13,98 @@
...
+
+	~UTransportMock()
+    {
+        mock_info.deinit_passed = true;     
+    }
+
+private:
+	[[nodiscard]] v1::UStatus sendImpl(const v1::UMessage& message) override {
+        v1::UStatus retval;
+        mock_info.payload = message.payload();
```
Comment (gregmedd) [inline test/include/UTransportMock.h:85]: It might be helpful to capture the whole message

File: test/include/UTransportMock.h [hunk 2]
```diff
@@ -13,5 +13,98 @@
...
+        retval.set_message(mock_info.send_status_message);
+        mock_info.send_count++;
+        return retval;
+	}
+
+	[[nodiscard]] v1::UStatus registerListenerImpl(
+	    const v1::UUri& sink_filter, CallableConn&& listener,
+	    std::optional<v1::UUri>&& source_filter) override {
+        mock_info.listener = listener;
+        mock_info.source_filter = source_filter;
```
Comment (gregmedd) [inline test/include/UTransportMock.h:96]: Also capture the sink_filter:
```suggestion
        mock_info.source_filter = source_filter;
        mock_info.sink_filter = sink_filter;
```

File: test/include/UTransportMock.h [hunk 3]
```diff
@@ -13,5 +13,98 @@
...
+        v1::UStatus retval;
+        retval.set_code(v1::UCode::OK);
+        return retval;
+	}
+
+	void cleanupListener(CallableConn listener) override
+    {
+        mock_info.cleanup_passed = true;      
+    }
```
Comment (gregmedd) [inline test/include/UTransportMock.h:105]: Capture the listener here, too:
```suggestion
	void cleanupListener(CallableConn listener) override
    {
        mock_info.cleanup_passed = true;
        mock_info.cleanup_listener = listener;   
    }
```

**Review by gregmedd — COMMENTED** [review]
So far the mock looks pretty good. I added a couple comments on the things that stood out to me immediately.

File: test/include/UTransportMock.h (hunk 1 above)
Comment (debruce) [reply test/include/UTransportMock.h:85]: I've changed the MockInfo class to store the entire message, and modified the test to just check the payload part of this. I will probably extend the checking of the stored packet in a future push.

File: test/include/UTransportMock.h (hunk 2 above)
Comment (debruce) [reply test/include/UTransportMock.h:96]: I've added the sink_filter to the capture. Will add checking in test layer in future push.

**Comment by gregmedd** [comment]
This will get us started and test authors can always update it should they need extra capabilities.

**Pull request merged**